libp2p/network/network.go

```diff
@@ -0,0 +1,219 @@
+package network
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+
+	"github.com/pokt-network/pocket/libp2p/identity"
+	"github.com/pokt-network/pocket/libp2p/protocol"
+	"github.com/pokt-network/pocket/p2p/providers"
+	"github.com/pokt-network/pocket/p2p/types"
+	poktCrypto "github.com/pokt-network/pocket/shared/crypto"
+	"github.com/pokt-network/pocket/shared/modules"
+)
+
+type libp2pNetwork struct {
+	logger      *modules.Logger
+	bus         modules.Bus
+	host        host.Host
+	topic       *pubsub.Topic
+	addrBookMap types.AddrBookMap
+}
+
+var (
+	ErrNetwork = types.NewErrFactory("LibP2P network error")
+	Year       = time.Hour * 24 * 365
+	// TECHDEBT: consider more carefully and parameterize.
+	DefaultPeerTTL = 2 * Year
+)
+
+// TECHDEBT: factor out args which are common to network
+// implementations to an options or config struct.
+func NewLibp2pNetwork(
+	bus modules.Bus,
+	addrBookProvider providers.AddrBookProvider,
+	currentHeightProvider providers.CurrentHeightProvider,
+	log *modules.Logger,
+	host_ host.Host,
```

Comment:
I would rename the import to `hostLibP2P` and just use `host` directly as a varname